File: src/p0019/cpp/solution.cpp
class Solution {
public:
    ListNode *removeNthFromEnd(ListNode *head, int n) {
        ListNode sentinel(0);
        sentinel.next = head;
        auto first = &sentinel;
        for (int i = 0; i < n; i++) {
            first = first->next;
        }
        auto second = &sentinel;
        while (first->next) {
            first = first->next;
            second = second->next;
        }
        auto removeNode = second->next;
        second->next = removeNode->next;
//        delete removeNode;
        return sentinel.next;
    }
};

File: src/p0019/golang/solution.go
func removeNthFromEnd(head *ListNode, n int) *ListNode {
	sentinel := ListNode{Next: head}
	first := &sentinel
	for i := 0; i < n; i++ {
		first = first.Next
	}
	second := &sentinel
	for first.Next != nil {
		first = first.Next
		second = second.Next
	}
	removeNode := second.Next
	second.Next = removeNode.Next
	return sentinel.Next
}
